### DISCUSSION TITLE
Should `azd up` include `azd init`?

### DISCUSSION THREAD

**Discussion by savannahostrowski** [Azure/azure-dev#114] [General]
Right now, `azd` has a single command (`azd up`) that combines `azd init`, `azd provision` and `azd deploy`.

We've received some developer feedback that bundling "getting code" (via `init`) and "getting my app running on Azure" (`provision` and `deploy`) in a single command is confusing. Largely, this seems to be because developers almost always:

Get code (clone repo etc.) -> modify -> deploy app

We're interested in hearing your thoughts! Is this confusing to you?

**Comment by mhassaninmsft** [comment]
yes I agree with this, since besides demo and hello world applications, almost always I want to modify the scaffolding and some infrastructure code. perhaps `azd up`. As it is right now, the command is a little confusing

**Comment by TWolverson** [comment]
Yes, I agree that it is confusing, maybe primarily because `up` is described in the documentation as the principal command to use. Besides my first time using the tool or using the tool with a new template, where I might want to try the sample unaltered, deploying an unmodified template to my Azure environment won't be useful. `up` could still be useful solely in the context of trying out a new template, but the principal commands for every day real work would be `init`/`provision`/`deploy` and those should be front and centre.

**Comment by savannahostrowski** [comment]
Following up here, the latest release of `azd` (0.8.0), removes `init` from `up`!
